uvicorn/protocols/http/h11_impl.py

```diff
@@ -385,25 +420,29 @@ async def run_asgi(self, app):
                 self.logger.error(msg)
                 self.transport.close()
         finally:
-            self.on_response = None
+            self.on_response = lambda: None
 
-    async def send_500_response(self):
+    async def send_500_response(self) -> None:
         await self.send(
-            {
-                "type": "http.response.start",
-                "status": 500,
-                "headers": [
+            HTTPResponseStartEvent(
+                type="http.response.start",
+                status=500,
+                headers=[
                     (b"content-type", b"text/plain; charset=utf-8"),
                     (b"connection", b"close"),
                 ],
-            }
+            )
         )
         await self.send(
-            {"type": "http.response.body", "body": b"Internal Server Error"}
+            HTTPResponseBodyEvent(
```

Comment:
Should we just annotate, or is it fine to instantiate the dictionary with this TypedDic?

Reply:
If you can get it to work with a plain dict, I'd prefer that, I imagine there is _some_ overhead to using the TypedDict